ksonnet-gen/ksonnet/catalog.go

```diff
@@ -50,9 +52,22 @@ func NewCatalog(apiSpec *spec.Swagger, opts ...CatalogOpt) (*Catalog, error) {
 		return nil, errors.New("apiSpec is nil")
 	}
 
+	if apiSpec.Info == nil {
+		return nil, errors.New("apiSpec Info is nil")
+	}
+
+	parts := strings.SplitN(apiSpec.Info.Version, ".", 3)
```

Comment:
Is there any possibility that this will split into 2 parts instead of 3, i.e. `1.8`? May want to add error checking.